types/schemas/postSubmissionApplication/index.ts

```diff
@@ -1,57 +1,34 @@
-import {
-  AdvertConsentApplicationType,
-  AmendmentApplicationType,
-  ApplicationType,
-  ApprovalApplicationType,
-  ComplianceConfirmationApplicationType,
-  EnvironmentalImpactApplicationType,
-  HazardousSubstanceConsentApplicationType,
-  HedgerowRemovalNoticeApplicationType,
-  LandDrainageConsentApplicationType,
-  LDCApplicationType,
-  ListedApplicationType,
-  NotifyCompletionApplicationType,
-  ObligationApplicationType,
-  OnshoreExtractionOilAndGasApplicationType,
-  PAApplicationType,
-  PPApplicationType,
-  PrimaryApplicationType,
-  RightsOfWayOrderApplicationType,
-  WTTApplicationType,
-} from '../prototypeApplication/enums/ApplicationType';
 import {PrototypeApplication} from '../prototypeApplication';
+import {ApplicationType} from '../prototypeApplication/enums/ApplicationType';
 import {PostSubmissionMetadata} from './Metadata';
+import {Appeal} from './data/Appeal';
+import {Application} from './data/Application';
+import {Assessment} from './data/Assessment';
 import {CaseOfficer} from './data/CaseOfficer';
 import {PublicComments, SpecialistComments} from './data/Comment';
-import {Submission} from './data/Submission';
-import {Validation} from './data/Validation';
 import {Consultation} from './data/Consultation';
-import {Assessment} from './data/Assessment';
-import {Appeal} from './data/Appeal';
-import {Application} from './data/Application';
 import {LocalPlanningAuthority} from './data/LocalPlanningAuthority';
+import {Submission} from './data/Submission';
+import {Validation} from './data/Validation';
+
 /**
  * @internal
  * The generic base type for all applications
  * Takes a primary and granular application type which allows child properties to differ based on these inputs
- * Deriving `TPrimary` from `TGranular` is possible in TS, but not in a way which is currently compatible with ts-json-schema-generator
  *
  * @todo create redacted form of PrototypeApplication
  */
-interface ApplicationSpecification<
-  TPrimary extends PrimaryApplicationType,
-  TGranular extends ApplicationType,
-> {
-  applicationType: TGranular;
+interface ApplicationSpecification<T extends ApplicationType> {
+  applicationType: T;
```

Comment:
This is the main change ! `T` is now always a granular `ApplicationType` enum value